app/models/concerns/deputy_user.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module DeputyUser
+  extend ActiveSupport::Concern
+
+  included do
+    # rubocop:disable Rails/InverseOf
+    belongs_to :deputy,
+               class_name: 'User',
+               foreign_key: :deputy_user_id,
+               optional: true
+    # rubocop:enable Rails/InverseOf
+
+    validate :deputy_must_be_assigned
+  end
+
+  private
+
+    # @note We can't tell (easily) if the deputy is associated with the exact user, but we can tell if they have any
+    # valid primary users.
+    def deputy_must_be_assigned
+      return if deputy.nil?
+
+      errors.add(:deputy, :not_assigned) if deputy.deputy_assignments.limit(1).empty?
```

Comment:
Is this better than `deputy.deputy_assignments.none?`?

Reply:
I'll take a look at the sql that's generated. I don't have a preference for either, but `none?` does read better.

Reply:
In looking at the logs, calling `none?` instead of `limit(1)` produces NO sql output. It works, but I'm not seeing any indication that it's looking at the database. So I gather it must be using some kind of cached reference? Which I guess is ok? 🤷

Reply:
Interesting. I don't have a strong preference here. I was just curious.

Reply:
`.none?` is the better option here. Ultimately the SQL produced is very similar: 

```sql
# deputy.deputy_assignments.none?
SELECT 1 AS one FROM "deputy_assignments" WHERE "deputy_assignments"."deputy_user_id" = $1 LIMIT $2
```

but as you can see from the `SELECT 1`, AR will not actually load anything out of the database or instantiate an AR object. That's negligibly more efficient (probably doesn't matter) and more readable (does matter)